File: 0_Trabalho_Pratico_Padaria/Bakehouse/Bakehouse.Application/ViewObjects/Category/CategoryVO.cs
using FluentResults;
using System;

namespace Bakehouse.App.ViewObjects.Category
{
    public class CategoryVO
    {
        public int Id { get; set; }
        public string Description { get; set; }
        public DateTime CreatedAt { get; set; }

        public Result Valid()
        {
            if (Description.Trim().Length <= 0)
                return Result.Fail("Descrição da categoria deve ser maior que zero");

            return Result.Ok();
        }
    }
}

File: 0_Trabalho_Pratico_Padaria/BakehouseJpaHB/src/main/java/com/bakehouse/viewobjects/category/CategoryVO.java
package com.bakehouse.viewobjects.category;

import com.bakehouse.helpers.Result;
import com.bakehouse.helpers.Validations;

public class CategoryVO {
    private int id;
    private String description;

    public CategoryVO() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
    
    public Result valid() {
        try {
            if (id < -1 || id == 0)
                return new Result("Id inválido", false);
            if (Validations.stringIsNullOrEmpty(description))
                return new Result("Descrição não informada", false);
            
            return new Result("OK", true);
        } catch (Exception ex) {
            return new Result("Ocorreu um erro inesperado", false);
        }
    }
}